packages/core/src/routes/role.application.ts

```diff
@@ -0,0 +1,129 @@
+import { Applications } from '@logto/schemas';
+import { generateStandardId } from '@logto/shared';
+import { tryThat } from '@silverhand/essentials';
+import { object, string } from 'zod';
+
+import RequestError from '#src/errors/RequestError/index.js';
+import koaGuard from '#src/middleware/koa-guard.js';
+import koaPagination from '#src/middleware/koa-pagination.js';
+
+import type { AuthedRouter, RouterInitArgs } from './types.js';
+
+export default function roleApplicationRoutes<T extends AuthedRouter>(
+  ...[router, { queries }]: RouterInitArgs<T>
+) {
+  const {
+    roles: { findRoleById },
+    applications: { countM2mApplicationsByIds, findM2mApplicationsByIds, findApplicationById },
+    applicationsRoles: {
+      findFirstApplicationsRolesByRoleIdAndApplicationIds,
+      findApplicationsRolesByRoleId,
+      insertApplicationsRoles,
+      deleteApplicationRole,
+    },
+  } = queries;
+
+  router.get(
+    '/roles/:id/applications',
+    koaPagination(),
+    koaGuard({
+      params: object({ id: string().min(1) }),
+      response: Applications.guard.array(),
+      status: [200, 204, 400, 404],
+    }),
+    async (ctx, next) => {
+      const {
+        params: { id },
+      } = ctx.guard;
+      const { limit, offset } = ctx.pagination;
+
+      await findRoleById(id);
+
+      return tryThat(
+        async () => {
+          const applicationRoles = await findApplicationsRolesByRoleId(id);
+          const applicationIds = applicationRoles.map(({ applicationId }) => applicationId);
+
+          const [{ count }, applications] = await Promise.all([
+            countM2mApplicationsByIds(applicationIds),
+            findM2mApplicationsByIds(limit, offset, applicationIds),
+          ]);
+
+          ctx.pagination.totalCount = count;
+          ctx.body = applications;
+
+          return next();
+        },
+        (error) => {
```

Comment:
This "tryThat" looks like for parsing search fields only

Reply:
`parseSearchParamsForSearch`

Reply:
Fixed